packages/pyright-internal/src/analyzer/checker.ts

```diff
@@ -3307,6 +3318,48 @@ export class Checker extends ParseTreeWalker {
                 }
             }
         });
+
+        // based overload consistency part 2: Implementation return type must be a subtype of the union of overload return types.
+        if (
+            this._importResolver.getConfigOptions().overloadConsistencyStrictness === 'strict' &&
+            implementation &&
+            isFunction(implementation)
+        ) {
+            const implNode = implementation.shared.declaration?.node?.parent;
+            let implBound = implementation;
+            if (implNode) {
+                const liveScopeIds = ParseTreeUtils.getTypeVarScopesForNode(implNode);
+                implBound = makeTypeVarsBound(implementation, liveScopeIds);
+            }
+
+            const implReturnType =
+                FunctionType.getEffectiveReturnType(implBound) ?? this._evaluator.getInferredReturnType(implBound);
```

Comment:
i think you want this
```suggestion
                this._evaluator.getEffectiveReturnType(implBound);
```

Reply:
i was just coppying the rest of the traut code here

Reply:
I suspect that code might have been written before this method existed? Idk

Reply:
weird. it's private in `typeEvaluator.ts` but i have no idea why